src/deferred.js

```diff
@@ -316,14 +314,17 @@ jQuery.extend( {
 					if ( values === progressValues ) {
 						master.notifyWith( contexts, values );
 					} else if ( !( --remaining ) ) {
-						master.resolveWith( contexts, values );
+						master.resolveWith(
+							contexts.length === 1 ? contexts[ 0 ] : contexts,
+							values
+						);
 					}
 				};
 			},
 			progressValues, progressContexts, resolveContexts;
 
 		// Add listeners to Deferred subordinates; treat others as resolved
-		if ( length > 1 ) {
+		if ( length > 0 ) {
 			progressValues = new Array( length );
 			progressContexts = new Array( length );
 			resolveContexts = new Array( length );
```

Comment:
Ah, yes, that would be a change in behavior requiring tests. If we reverse them, will the `promise` case ever get used – since Deferred has `.then`?